### ISSUE TITLE
Model Sensititivity Aanlysis

### ISSUE THREAD

**Issue by ALearningCurve** [ALearningCurve/ticket-forge#79]
From section 5 of the requirements document.

Perform sensitivity analysis to determine how the model's performance  changes with respect to different 
- **feature importance** for the selected model (SHAP or LIME) to understand how features impact model predictions
- **Hyperparam sensitivity** analyze how changes in hyperparameters affect model performance. Particularly intested in which hyperparameters have the biggest affect on outcomes

For the hyperparam, you can simply load the Sklearn gridsearch pickle or the results json to create a graphic after the training script runs.

In the past I have done something similar for hyperparam sensitivity by plotting performance on y axis and the hyperparam value on the x axis for each training run:
<img width="1200" height="1600" alt="Image" src="https://github.com/user-attachments/assets/57257c47-dafa-4b80-bcf9-5b0c0abfa285" />
